### PR TITLE
[core/os] get_current_directory: Add allocator arg to targets where i…

### PR DESCRIPTION
**Pull request by greenya**
At the moment, `os.get_current_directory` has different signature on different targets. Only Windows and JS targets has `allocator := context.allocator` argument; all other targets missing it, leading to compilation error of user code if it uses that argument on other targets.

**This PR adds `allocator` argument to implementation of `os.get_current_directory` on targets where it is missing.**

### CODE REVIEW

**Comment by Feoramund** [comment]
`context.allocator` must be assigned to with the allocator, or the allocator must be used in relevant allocating procedures for this to work as intended; merely adding the argument does nothing.

**Comment by greenya** [comment]
The PR fixes compilation errors, which are due to different proc signatures. If user code has line like this:
```odin
log.debugf("current dir is %v", os.get_current_directory(context.temp_allocator))
```
It will compile on Windows and not on Mac (and other targets). To make it compilable, user have to check target with construction like `when ODIN_OS == .Windows {`. I don't think its intended.

**Comment by jasonKercher** [comment]
What Feormund is saying that just because the parameter exists doesn't mean it is being used.  In your example, you send `context.temp_allocator` to `os.get_current_directory`, but that is not going to cause `get_current_directory` to use it.

File: core/os/os_darwin.odin [hunk 1]
```diff
@@ -1095,7 +1095,7 @@ unset_env :: proc(key: string) -> Error {
 }
 
 @(require_results)
-get_current_directory :: proc() -> string {
+get_current_directory :: proc(allocator := context.allocator) -> string {
 	page_size := get_page_size() // NOTE(tetra): See note in os_linux.odin/get_current_directory.
```
Comment (laytan) [inline core/os/os_darwin.odin:1099]:
```suggestion
	context.allocator = allocator
	page_size := get_page_size() // NOTE(tetra): See note in os_linux.odin/get_current_directory.
```

File: core/os/os_freebsd.odin [hunk 2]
```diff
@@ -840,7 +840,7 @@ get_env :: proc(key: string, allocator := context.allocator) -> (value: string)
 }
 
 @(require_results)
-get_current_directory :: proc() -> string {
+get_current_directory :: proc(allocator := context.allocator) -> string {
 	// NOTE(tetra): I would use PATH_MAX here, but I was not able to find
```
Comment (laytan) [inline core/os/os_freebsd.odin:844]:
```suggestion
	context.allocator = allocator
	// NOTE(tetra): I would use PATH_MAX here, but I was not able to find
```

File: core/os/os_linux.odin [hunk 3]
```diff
@@ -985,7 +985,7 @@ unset_env :: proc(key: string) -> Error {
 }
 
 @(require_results)
-get_current_directory :: proc() -> string {
+get_current_directory :: proc(allocator := context.allocator) -> string {
 	// NOTE(tetra): I would use PATH_MAX here, but I was not able to find
```
Comment (laytan) [inline core/os/os_linux.odin:989]:
```suggestion
	context.allocator = allocator
	// NOTE(tetra): I would use PATH_MAX here, but I was not able to find
```

File: core/os/os_netbsd.odin [hunk 4]
```diff
@@ -894,7 +894,7 @@ get_env :: proc(key: string, allocator := context.allocator) -> (value: string)
 }
 
 @(require_results)
-get_current_directory :: proc() -> string {
+get_current_directory :: proc(allocator := context.allocator) -> string {
 	// NOTE(tetra): I would use PATH_MAX here, but I was not able to find
```
Comment (laytan) [inline core/os/os_netbsd.odin:898]:
```suggestion
	context.allocator = allocator
	// NOTE(tetra): I would use PATH_MAX here, but I was not able to find
```

File: core/os/os_openbsd.odin [hunk 5]
```diff
@@ -806,7 +806,7 @@ get_env :: proc(key: string, allocator := context.allocator) -> (value: string)
 }
 
 @(require_results)
-get_current_directory :: proc() -> string {
+get_current_directory :: proc(allocator := context.allocator) -> string {
 	buf := make([dynamic]u8, MAX_PATH)
```
Comment (laytan) [inline core/os/os_openbsd.odin:810]:
```suggestion
	context.allocator = allocator
	buf := make([dynamic]u8, MAX_PATH)
```

**Pull request merged**